.github/workflows/pr-test.yml

```diff
@@ -43,16 +43,25 @@ jobs:
         uses: tj-actions/changed-files@eaf854ef0c266753e1abec356dcf17d92695b251 # v44
         with:
           files: tests/cloudformation/checks/resource/aws/**/*
-      - name: Install cfn-lint
+      - name: Filter YAML and JSON files
         if: steps.changed-files-specific.outputs.any_changed == 'true'
+        id: filter-files
+        run: |
+          YAML_JSON_FILES=$(echo ${{ steps.changed-files-specific.outputs.all_changed_files }} | tr ' ' '\n' | grep -E '\.ya?ml$|\.json$' | tr '\n' ' ')
+          if [ -n "$YAML_JSON_FILES" ]; then
+            echo "YAML_JSON_FILES=$YAML_JSON_FILES" >> "$GITHUB_ENV"
+            echo "RELEVANT_FILES_CHANGED=true" >> "$GITHUB_ENV"
+          else
+            echo "RELEVANT_FILES_CHANGED=false" >> "$GITHUB_ENV"
+          fi
+      - name: Install cfn-lint
+        if: env.RELEVANT_FILES_CHANGED == 'true'
         run: |
           pip install -U cfn-lint
       - name: Lint Cloudformation templates
-        if: steps.changed-files-specific.outputs.any_changed == 'true'
-        env:
-          ALL_CHANGED_FILES: ${{ steps.changed-files-specific.outputs.all_changed_files }}
+        if: env.RELEVANT_FILES_CHANGED == 'true'
```

Comment:
Do we need to check this again?
Can this be run in the previous step where we already checked?

Reply:
you're right, the 2 steps can be merged